x/group/keeper/msg_server.go

```diff
@@ -373,9 +374,14 @@ func (k Keeper) CreateGroupPolicy(goCtx context.Context, msg *group.MsgCreateGro
 	// loop here in the rare case where a ADR-028-derived address creates a
 	// collision with an existing address.
 	for {
-		nextAccVal := k.groupPolicySeq.NextVal(ctx.KVStore(k.key))
-		derivationKey := make([]byte, 8)
-		binary.BigEndian.PutUint64(derivationKey, nextAccVal)
+		// TODO HV2: this code should work, but I believe it breaks the intended functionality of the group module.
+		//  This implementation is here only for compatibility and allow the tests to pass.
+		//  Otherwise, NewBaseAccountWithPubKey(pubKey) would fail with any derivationKey.
+		//  This is because it calls NewBaseAccountWithAddress, which validates the address to be ethereum hex compatible
+		//  We are not going to use group module (as we don't use multisig or accounts policies).
+		//  However, is there a better alternative which keeps the group module operational?
```

Comment:
If Heimdall won't use the group module, how come we are hitting this code? Is it only hit by unit tests?

Reply:
it was to ensure `NewBaseAccountWithAddress` in `auth` module is always working as expected.
Since we are not using this module, I'll revert the changes

Reply:
Resolved